lang/cs/Org.Apache.REEF.Wake/Remote/IWritable.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Interface that classes should implement if they need to be readable to and writable 
+    /// from the stream. It is assumed that the classes inheriting this interface will have a 
+    /// default empty constructor
+    /// </summary>
+    public interface IWritable
+    {
+        /// <summary>
+        /// Read the class fields from the stream.
+        /// </summary>
+        /// <param name="stream">The stream from which to read</param>
+        /// <param name="optionalParameters">The optional parameters to be passed to the reader.
```

Comment:
Yes...we need optional parameters I think either in DefaultRemoteManager or
in NsMessage. I did not have it earlier but then later felt the necessity

On Thu, Apr 16, 2015 at 4:13 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Wake/Remote/IWritable.cs
> <https://github.com/apache/incubator-reef/pull/149#discussion_r28558512>:
>
> > +using System.Threading.Tasks;
> > +
> > +namespace Org.Apache.REEF.Wake.Remote
> > +{
> > +    /// <summary>
> > +    /// Interface that classes should implement if they need to be readable to and writable
> > +    /// from the stream. It is assumed that the classes inheriting this interface will have a
> > +    /// default empty constructor
> > +    /// </summary>
> > +    public interface IWritable
> > +    {
> > +        /// <summary>
> > +        /// Read the class fields from the stream.
> > +        /// </summary>
> > +        /// <param name="stream">The stream from which to read</param>
> > +        /// <param name="optionalParameters">The optional parameters to be passed to the reader.
>
> Do we need those?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/149/files#r28558512>.
>